In [45]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [46]:
#Read file
data=pd.read_csv('/content/StudentsPerformance.csv')

In [6]:
#print the given data
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [ ]:
#   1. Find out how many males and females participated in the test?

In [47]:
#Finding the value counts in gender category.So we can find number of males and females.
data['gender'].value_counts()
  

female    518
male      482
Name: gender, dtype: int64

In [ ]:
# 2. What do you think about the students' parental level of education?

In [8]:
# Finding the value counts for parental level of education.
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Based on the output 226 parents went to some college.222 parents having associate's degree.196 parents having high school education.179 parents having some high school education. 118 parents having bachelor's degree and only 59 parents having master's degree.From this we can conclude that most of them having a college degree. Only 37.5% parents having high school education.Majority of the parents are well educated. 

In [ ]:
#3a.Who scores the most on average for math, reading and writing based on gender

In [17]:
df1=data.loc[data['gender'] == 'female', 'math score'].mean()
df2=data.loc[data['gender'] == 'male', 'math score'].mean()
print("Average of math score\n")
print('Female:\t',df1)
print('Male:\t',df2)

Average of math score

Female:	 63.37644787644788
Male:	 68.82157676348548


Males having highest average

In [19]:
df3=data.loc[data['gender'] == 'female', 'reading score'].mean()
df4=data.loc[data['gender'] == 'male', 'reading score'].mean()
print("Average of reading score\n")
print('Female:\t',df3)
print('Male:\t',df4)

Average of reading score

Female:	 72.59073359073359
Male:	 65.54564315352697


Girls having highest average

In [21]:
df5=data.loc[data['gender'] == 'female', 'writing score'].mean()
df6=data.loc[data['gender'] == 'male', 'writing score'].mean()
print("Average of writing score\n")
print('Female:\t',df5)
print('Male:\t',df6)

Average of writing score

Female:	 72.46718146718146
Male:	 63.446058091286304


Females having highest average

In [26]:
# First we want to know the different types includedin test preparation course. From finding value count we get this consist of only 2 type ,ie 'none' & 'completed'

data['test preparation course'].value_counts()

none         642
completed    358
Name: test preparation course, dtype: int64

In [ ]:
#3b.Who scores the most on average for math, reading and writing based on test preparation course

In [28]:
df7=data.loc[data['test preparation course'] == 'none', 'math score'].mean()
df8=data.loc[data['test preparation course'] == 'completed', 'math score'].mean()
print("Average of math score\n")
print('None:\t',df7)
print('Completed:\t',df8)


Average of math score

None:	 63.78971962616822
Completed:	 69.96648044692738


In [30]:
df9=data.loc[data['test preparation course'] == 'none', 'reading score'].mean()
df10=data.loc[data['test preparation course'] == 'completed', 'reading score'].mean()
print("Average of reading score\n")
print('None:\t',df9)
print('Completed:\t',df10)


Average of reading score

None:	 66.41744548286604
Completed:	 74.17597765363128


In [33]:
df11=data.loc[data['test preparation course'] == 'none', 'writing score'].mean()
df12=data.loc[data['test preparation course'] == 'completed', 'writing score'].mean()
print("Average of writing score\n")
print('None:\t',df11)
print('Completed:\t',df12)

Average of writing score

None:	 64.45794392523365
Completed:	 74.68435754189944


In [ ]:
#4a.What do you think about the scoring variation for math, reading and writing based on Gender

In [35]:
df13=data.loc[data['gender'] == 'female', 'math score'].var()
df14=data.loc[data['gender'] == 'male', 'math score'].var()
print("Variance of math score\n")
print('Female:\t',df13)
print('Male:\t',df14)

Variance of math score

Female:	 256.9585931607208
Male:	 211.88909688494752


In [36]:
df15=data.loc[data['gender'] == 'female', 'reading score'].var()
df16=data.loc[data['gender'] == 'male', 'reading score'].var()
print("Variance of reading score\n")
print('Female:\t',df15)
print('Male:\t',df16)

Variance of reading score

Female:	 207.67743814552327
Male:	 200.2110100844541


In [37]:
df17=data.loc[data['gender'] == 'female', 'writing score'].var()
df18=data.loc[data['gender'] == 'male', 'writing score'].var()
print("Variance of writing score\n")
print('Female:\t',df17)
print('Male:\t',df18)

Variance of writing score

Female:	 220.36932705017813
Male:	 202.4139241379905


In [ ]:
#4b.What do you think about the scoring variation for math, reading and writing based on Test preparation course

In [38]:
df19=data.loc[data['test preparation course'] == 'none', 'math score'].var()
df20=data.loc[data['test preparation course'] == 'completed', 'math score'].var()
print("Variance of math score\n")
print('None:\t',df19)
print('Completed:\t',df20)


Variance of math score

None:	 246.668661699739
Completed:	 210.88402735395837


In [40]:
df21=data.loc[data['test preparation course'] == 'none', 'reading score'].var()
df22=data.loc[data['test preparation course'] == 'completed', 'reading score'].var()
print("Variance of reading score\n")
print('None:\t',df21)
print('Completed:\t',df22)

Variance of reading score

None:	 213.41985118657084
Completed:	 183.2658638874544


In [41]:
df23=data.loc[data['test preparation course'] == 'none', 'writing score'].var()
df24=data.loc[data['test preparation course'] == 'completed', 'writing score'].var()
print("Variance of writing score\n")
print('None:\t',df23)
print('Completed:\t',df24)

Variance of writing score

None:	 226.25173866767753
Completed:	 175.20261177096538


In [ ]:
#5.The management needs your help to give bonus points to the top 25% of students based on their math score, so how will you help the management to achieve this.

In [44]:
new_data= data.sort_values(['math score'], ascending=[False])
new_data.head(250)

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]